### DISCUSSION TITLE
Contributions (Commits) are not being displayed in github profile

### DISCUSSION THREAD

**Discussion by Ash-2k3** [community/community#50162] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Hey folks...I have been contributing in oppia for past few months and I recently noticed that my commits of merged PR are not being displayed in contribution graph. Can anyone please help me with it.

My email in local pc's git configurations and github do match.

<img width="739" alt="Screenshot 2023-03-16 at 01 10 55" src="https://user-images.githubusercontent.com/98253080/225424822-5f6bb42e-3593-47fb-a7e9-eaaf2fbbfcb4.png">
<img width="1440" alt="Screenshot 2023-03-16 at 01 09 41" src="https://user-images.githubusercontent.com/98253080/225424852-5a8c47d5-c0f8-4db0-95f9-9aeaa5470323.png">
Thanks

**Comment by airtower-luna** [comment, marked as answer]
Can you link one of the affected commits? That is, a commit that you think should show up in your contribution graph but doesn't.

Also, the screenshot shows that you have "keep my email address private" enabled but your email address is clearly visible in the screenshot. Are you sure that's what you want? :sweat_smile:

**Reply by Ash-2k3** [reply]
I think commits of this [PR](https://github.com/oppia/oppia/pull/17408) might be one of affected ones. 

Well what's done is done....I am lazy to change the ss now 👀

**Reply by airtower-luna** [reply]
The commit https://github.com/oppia/oppia/commit/5ab62843be6180ea3185f561e5fb10a2702b4e52 does show up in your contribution history, on February 25th, also that you made the PR on the 19th.

It looks like the PR was "merged" with the "squash" method, which instead of a proper Git merge creates one new commit combining all the changes of all the commits in the PR. As a result only that one new commit will show in your contribution history.

**Reply by Ash-2k3** [reply]
oh okay, I didn't know that squash merge will show only one new commit in contribution history.
Thanks for help @airtower-luna

**Comment by Kibet-mutai** [comment]
I'm experiencing the same problem. Contribution are not being displayed in my github profile after pushing commits.

**Comment by sushmitha047** [comment]
Same here. I am experiencing this issue for at least a week now.

**Comment by airtower-luna** [comment]
@Kibet-mutai, @Sushmitha047, looking at your most recently updated repos you both have the same problem: The `user.email` isn't set correctly, so the author email in your commits isn't one GitHub recognizes. It needs to be set to an email address connected to your account for the commits to be recognized as yours, see: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git

**Reply by Kibet-mutai** [reply]
The problem is I have to set `user.email `everytime before making commits.

**Reply by airtower-luna** [reply]
Do you frequently use different computers or something like that? The `user.email` setting should be persistent in your Git config.

**Reply by Kibet-mutai** [reply]
No, I use one computer.

**Reply by airtower-luna** [reply]
Exactly how (which command) do you set your user email?

**Reply by Kibet-mutai** [reply]
git config --global user.email "email"

**Reply by airtower-luna** [reply]
That command is correct, and should definitely stick. I don't know your setup, but I'm afraid you'll have to look what else (maybe another program?) might change or delete your Git config.

**Comment by EMakarovas** [comment]
I seem to be having the same issue, as far as I can tell all the conditions are met (the email address is associated with my account, the commits have the said email address, and I am showing up as a contributor in that repository; I have even starred the repo since [this link](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile#your-local-git-commit-email-isnt-connected-to-your-account) mentions it).

Example commit: https://github.com/sbuslab/utils/commit/31f8f836a0c71cf63bd4c5cdb2d6511535fe49cc
Email address: edgaras.makarovas@copper.co

Any ideas what might be missing?

**Comment by NataliaCzeladka** [comment]
I have the same issue. My commits do not register on the contribution graph, but I can see them all in the repository. 
The email address matches. Any ideas of what I am doing wrong? I've spent hours trying to resolve this problem and still nothing.

**Reply by airtower-luna** [reply]
Looking at your pinned repository: The email address in your commits is actually *empty*, see the `From` line [here](https://github.com/NataliaCzeladka/celestial-buddies/commit/ef2a1c6733d4e4da4736cdcb5b2c0d1700fc3786.patch) for example. Make sure to [set it to an address connected to your account](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address).

**Reply by NataliaCzeladka** [reply]
Thank you very much for checking this for me! I'm a complete newbie working on my first project ever. I think my email address didn't show because of my privacy settings. I've changed them now, so it should all work :)``

**Reply by airtower-luna** [reply]
Changing your privacy settings won't change anything, you need to set the correct `user.email` in your local Git config, as described in the documentation I linked.

**Reply by NataliaCzeladka** [reply]
I've done it before, but the email address I set there was my private email, not the one provided by github, so I assumed it might have not shown, because I had "Keep my email address private" box ticked. I had the same email address set in my github account and in Git from the very beginning. That's why I couldn't work out what went wrong. The funny thing is my first 2 or 3 commits do show in github and I haven't changed anything since then.

**Reply by airtower-luna** [reply]
> I've done it before, but the email address I set there was my private email, not the one provided by github, so I assumed it might have not shown, because I had "Keep my email address private" box ticked.

No, in that case the attempt push the commit(s) should be rejected. It's not about *not showing*, it's that the author address recorded in the commit is empty.

**Comment by mb010** [comment]
I can see the matching email in my commits from gitkraken, but they don't (and havent for a long time) shown up in my github contributions.

E.g.: [this commit](https://github.com/mb010/AstroAugmentations/commit/bff4a4a640bc4d1d46727a983f23ad851d6d7a82)

**Reply by airtower-luna** [reply]
Looks to me like there's a typo in the mail address, I'm pretty sure Manchester is spelled with only one "s". :wink:

**Reply by mb010** [reply]
The tragedy. :rofl: Welp. Haha. Thanks @airtower-luna !
